### PR TITLE
Fix/deposit service

### PR DESCRIPTION
**Pull request by Alejoboga20**
**Pull request type**

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Migration
- [ ] Other (please describe):

**What changes does this PR introduce?**

- Refactoring deposit service to match new requirements for circle 3ds verifications.
- Separating UserBalance section into small components.
- Adding customMsg to Notification Modal.
- Controlling redirection urls in topbar layout

Issue Link: https://rootstrap.atlassian.net/jira/software/projects/DOL/boards/93?selectedIssue=DOL-213

**Security Considerations**

- [ ] Perform server-side input validation and sanitization (whenever it's possible) on all user-controlled inputs.
- [ ] Perform context-aware output encoding/escaping of all user-controlled input.
- [ ] Enforce Access Control on every request. Consider the different types of users. Who should access that functionality/endpoint?
- [ ] Requires Test updates.

**Other information**

N/A

**Screenshots**
N/A

### CODE REVIEW

File: src/app/components/FundWalletModal/useFundWallet.ts [hunk 1]
```diff
@@ -45,6 +46,14 @@ export const useFundWallet = (open: boolean) => {
     }
   }, [isError, isSuccess, depositData]);
 
+  useEffect(() => {
+    if (isSuccess && depositData.result?.redirectUrl3ds && depositData.result?.paymentId) {
```
Comment (FlaviaCohen) [inline src/app/components/FundWalletModal/useFundWallet.ts:50]: It's a better practice to have this logic declared outside the useEffect and inside it only execute it.

File: src/app/components/Layout/TopBarLayout.tsx [hunk 2]
```diff
@@ -18,13 +20,35 @@ interface TopBarLayoutProps {
...
+    setIsNotificationModalOpen(false);
+    window.location.replace(window.location.origin);
+  };
+
   useEffect(() => {
     verifyStatus();
   }, [verifyStatus]);
 
+  useEffect(() => {
+    if (location.search === '?failure') {
```
Comment (FlaviaCohen) [inline src/app/components/Layout/TopBarLayout.tsx:41]: Same comment

File: src/app/components/NotificationModal/NotificationModal.tsx [hunk 3]
```diff
@@ -3,23 +3,29 @@ import { FailedVerification } from '../CC/Verification/FailedVerification';
...
+  customMsg = '',
+  handleClose,
+  isOpen,
+  isVerificationSuccess,
+}: NotificationModalProps) => {
   const t = useTranslation();
 
   return (
     <BaseModal open={isOpen} handleClose={handleClose}>
       <div style={{ textAlign: 'center' }}>
```
Comment (FlaviaCohen) [inline src/app/components/NotificationModal/NotificationModal.tsx:16]: Is it necessary to have this inline style? Can you move this to some stylesheet?

**Review by FlaviaCohen — CHANGES_REQUESTED** [review]

File: src/app/components/NotificationModal/NotificationModal.tsx (hunk 3 above)
Comment (Alejoboga20) [reply src/app/components/NotificationModal/NotificationModal.tsx:16]: sure I can move it

**Review by FlaviaCohen — APPROVED** [review]

**Pull request merged**